metadata-ingestion/src/datahub/ingestion/source/mongodb.py

```diff
@@ -467,9 +469,13 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
                     for mcp in MetadataChangeProposalWrapper.construct_many(
                         entityUrn=dataset_urn,
                         aspects=[
-                            schema_metadata,
-                            dataset_properties,
-                            data_platform_instance,
+                            aspect
+                            for aspect in [
+                                schema_metadata,
+                                dataset_properties,
+                                data_platform_instance,
+                            ]
+                            if aspect is not None
```

Comment:
`construct_many` automatically filters out None, so you don't need to do that here



```suggestion
                            schema_metadata,
                            dataset_properties,
                            data_platform_instance,
```